hail/src/main/scala/is/hail/expr/ir/EmitStream.scala

```diff
@@ -651,30 +651,33 @@ object EmitStream {
           val valueType = valueIR.pType
           val valueTI = coerce[Any](typeToTypeInfo(valueType))
           val vm = fb.newField[Boolean](name + "_missing")
-          val vv = fb.newField(name)(valueTI)
+          val vv = fb.newPField(name, valueType)
           val valuet = emitIR(valueIR, env)
-          val bodyEnv = env.bind(name -> ((valueTI, vm, vv)))
+          val bodyEnv = env.bind(name -> ((vm, vv.load())))
           emitStream(childIR, bodyEnv)
             .addSetup(_ => Code(
               valuet.setup,
               vm := valuet.m,
-              vv := vm.mux(defaultValue(valueType), valuet.v)))
+              vm.mux(
+                vv := valueType.defaultValue,
```

Comment:
Could you help me understand the benefit of this style (returning Code[Unit] from max, and assigning within the branches) vs the old (assigning from the return Code[_])

Reply:
defualtValue and valuet.pv are PValues.  I didn't write mux on PValues.  It would basically assign the PValue to a temporary (local or class) so this is more efficient.  All the uses of mux were on the right hand size of assignments like this.